code-challenges.test.js

```diff
@@ -15,47 +15,99 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
+describe("codedmessage"), () => {
+  it("takes in a string and returns a coded message"), () => {
+    const secretCodeWord1 = "Lackadaisical"
+    // // Expected output: "L4ck4d41s1c4l"
+    const secretCodeWord2 = "Gobbledygook"
+    // // Expected output: "G0bbl3dyg00k"
+    const secretCodeWord3 = "Eccentric"
+    // // Expected output: "3cc3ntr1c"
+expect(codedmessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l"),
+expect(codedmessage(secretCodeWord2)).toEqual("G0bbl3dyg00k"),
+expect(codedmessage(secretCodeWord3)).toEqual("3cc3ntr1c");
+  });
+});
 
 
 // b) Create the function that makes the test pass.
 
+const hackerSpeak = (statement) => {
+  // I am creating a function that will take a string and convert each letters selected with a number
+    const leet = {'a': 4, 'b': 3, 'i': 1, 'o': 0}
+    const speak = letter => leet.hasOwnProperty(letter) ? leet[letter] : letter
+
+    return statement.split('').map(speak).join('')
+// the .split will convert a string into an array, .map will do something to each elelmet of the array and the .join will bring it back into a string
+}
+
+
 
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe("particularletter", () => {
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter"), () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    // const letterA = "a"
+    // // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    // const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+    expect(particularletter(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"]),
+      expect(particularletter(arrayOfWords2)).toEqual(["Cherry", "Blueberry", "Peach"]);
+  });
+});
```

Comment:
The array of words looks great, but in this problem you will need to pass in two pieces of information into the function - the array and the letter.
`expect(particularletter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"]),`